features/step_definitions/collection.js

```diff
@@ -0,0 +1,273 @@
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 200);
+            Assert.deepEqual(response.body.pipelineIds, []);
+        });
+    });
+
+    When(/^they update the collection "myCollection" with that pipeline$/,
+        { timeout: TIMEOUT }, function step() {
+            return this.ensurePipelineExists({ repoName: this.repoName })
+                .then(() => {
+                    const pipelineId = parseInt(this.pipelineId, 10);
+
+                    request({
+                        uri: `${this.instance}/${this.namespace}/collections/` +
+                            `${this.firstCollectionId}`,
+                        method: 'PUT',
+                        auth: {
+                            bearer: this.jwt
+                        },
+                        body: {
+                            pipelineIds: [pipelineId]
+                        },
+                        json: true
+                    }).then((response) => {
+                        Assert.strictEqual(response.statusCode, 200);
+                    });
+                });
+        });
+
+    Given(/^they have a collection "myCollection"$/, function step() {
+        return createCollection.call(this, { name: 'myCollection' })
+            .then((response) => {
+                Assert.oneOf(response.statusCode, [409, 201]);
+
+                if (response.statusCode === 201) {
+                    this.firstCollectionId = response.body.id;
+                } else {
+                    const str = response.body.message;
+
+                    [, this.firstCollectionId] = str.split(': ');
+                }
+            });
+    });
+
+    Given(/^they have a collection "anotherCollection"$/, function step() {
+        return createCollection.call(this, { name: 'anotherCollection' })
+            .then((response) => {
+                Assert.oneOf(response.statusCode, [409, 201]);
+
+                if (response.statusCode === 201) {
+                    this.secondCollectionId = response.body.id;
+                } else {
+                    const str = response.body.message;
+
+                    [, this.secondCollectionId] = str.split(': ');
+                }
+            });
+    });
+
+    When(/^they fetch all their collections$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 200);
+            this.collections = response.body;
+        });
+    });
+
+    Then(/^they can see those collections$/, function step() {
+        Assert.strictEqual(this.collections.length, 2);
+        Assert.strictEqual(this.collections[0].name, 'anotherCollection');
```

Comment:
Do we need to check specifically against IDs? Should we just check if they have both `anotherCollection` and `myCollection`, not necessarily, that `anotherCollection` is ID 0 and `myCollection` is ID 1?

Reply:
I don't think we need to specifically check against IDs. I wrote it this way because the response sends them in descending order so if `anotherCollection` was created after `myCollection` it would be the first collection sent back but I'm not sure if that needs to strictly be checked against. I'll change it to just ensure that one of them is `anotherCollection` and the other is `myCollection`.